certgen/certProcessor/src/main/java/org/incredible/CertificateGenerator.java

```diff
@@ -42,33 +45,31 @@
 
     private CertificateFactory certificateFactory = new CertificateFactory();
 
-    private String directory;
+    private String directory = "conf/";
 
     public CertificateGenerator(Map<String, String> properties, String directory) {
         this.properties = properties;
         this.directory = directory;
     }
 
+    public CertificateGenerator(Map<String, String> properties) {
+        this.properties = properties;
+    }
 
-    public CertificateResponse createCertificate(CertModel certModel, HTMLTemplateProvider htmlTemplateProvider) throws
-            SignatureException.UnreachableException, InvalidDateFormatException, SignatureException.CreationException,
-            IOException, FontFormatException, NotFoundException, WriterException, StorageServiceException {
-        String uuid;
-        CertificateExtension certificateExtension = certificateFactory.createCertificate(certModel, properties);
-        String jsonData = generateCertificateJson(certificateExtension);
-        String accessCode = generateQrCodeForCertificate(certificateExtension);
-        String htmlContent = htmlTemplateProvider.getTemplateContent(directory);
-        if (htmlTemplateProvider.checkHtmlTemplateIsValid(htmlContent)) {
-            HtmlGenerator htmlGenerator = new HtmlGenerator(htmlContent);
-            htmlGenerator.generate(certificateExtension, directory);
-            uuid = getUUID(certificateExtension.getId());
-        } else {
-            return new CertificateResponse();
-        }
-        return new CertificateResponse(uuid, accessCode, jsonData, certModel.getIdentifier());
+    public CertificateExtension getCertificateExtension (CertModel certModel) throws SignatureException.UnreachableException,
```

Comment:
Earlier createCertificate was doing generating json, html-2-pdf and give back a response.
Now, i see those responsibilities lifted above. Can't it continue?

The idea is to make small and incremental changes to avoid regression and conflict code-design already in-place.

Reply:
Yes, previously 

certificateResponse = certificateGenerator.createCertificate(certModel, htmlTemplateZip);

createCertificate was doing QR code generation and uploading along with creating json file , html file and converting html to pdf.
but now we dont have resposibility of generating these files and its upload,its moved to print service , so now create certificate remains with only single responsibility i.e generation of QR image and its upload  which doesn't justify the createCertificate method name.